lib/install/phases/postsystem.go

```diff
@@ -104,6 +109,16 @@ func (p *waitExecutor) waitForAPI(ctx context.Context, done chan bool) {
 	}
 }
 
+func (p *waitExecutor) tryQueryAPI() error {
+	_, err := p.Client.CoreV1().ComponentStatuses().Get("scheduler", metav1.GetOptions{})
+	return trace.Wrap(err)
+}
+
+func (p *waitExecutor) tryQueryNamespace() error {
+	_, err := p.Client.CoreV1().Namespaces().Get(defaults.KubeSystemNamespace, metav1.GetOptions{})
```

Comment:
nit: I don't really an issue with this at this point, but it would make sense to look into informers and the wait functionality as implemented in client-go/kubectl.

Reply:
Yeah, it might be useful for a more generic scenario, I feel like it'd be an overkill in this case.